src/nix/export.cc

```diff
@@ -0,0 +1,126 @@
+    }
+
+    Examples examples() override
+    {
+        return {
+            Example{
+                "To export Hello:",
+                "nix export hello"
+            },
+        };
+    }
+
+    Strings getDefaultFlakeAttrPaths() override
+    {
+        Strings res{"defaultApp." + settings.thisSystem.get()};
+        for (auto & s : SourceExprCommand::getDefaultFlakeAttrPaths())
+            res.push_back(s);
+        return res;
+    }
+
+    Strings getDefaultFlakeAttrPathPrefixes() override
+    {
+        Strings res{"apps." + settings.thisSystem.get() + ".", "packages"};
+        for (auto & s : SourceExprCommand::getDefaultFlakeAttrPathPrefixes())
+            res.push_back(s);
+        return res;
+    }
+
+    void run(ref<Store> store) override
+    {
+        auto evalState = getEvalState();
+
+        auto app = installable->toApp(*evalState);
+        store->buildPaths(app.context);
+
+        auto [exporterFlakeRef, exporterName] = parseFlakeRefWithFragment(exporter, absPath("."));
+        const flake::LockFlags lockFlags{ .writeLockFile = false };
+        auto exporter = InstallableFlake(
+            evalState, std::move(exporterFlakeRef),
+            Strings{exporterName == "" ? ("defaultExporter." + settings.thisSystem.get()) : exporterName},
+            Strings({"exporters." + settings.thisSystem.get() + "."}), lockFlags);
+
+        Value * arg = evalState->allocValue();
+        evalState->mkAttrs(*arg, 1);
+
+        PathSet context;
+        for (auto & i : app.context)
+            context.insert("=" + store->printStorePath(i.path));
+        mkString(*evalState->allocAttr(*arg, evalState->symbols.create("program")), app.program, context);
+
+        auto vRes = evalState->allocValue();
+        evalState->callFunction(*exporter.toValue(*evalState).first, *arg, *vRes, noPos);
+
+        if (!evalState->isDerivation(*vRes))
+            throw Error("the exporter '%s' does not produce a derivation", exporter.what());
+
+        Bindings::iterator i = vRes->attrs->find(evalState->sDrvPath);
+        if (i == vRes->attrs->end())
+            throw Error("the exporter '%s' does not produce a derivation", exporter.what());
+
+        PathSet context2;
+        StorePath drvPath = store->parseStorePath(evalState->coerceToPath(*i->pos, *i->value, context2));
+
+        i = vRes->attrs->find(evalState->sOutPath);
+        if (i == vRes->attrs->end())
+            throw Error("the exporter '%s' does not produce a derivation", exporter.what());
+
+        StorePath outPath = store->parseStorePath(evalState->coerceToPath(*i->pos, *i->value, context2));
+
+        store->buildPaths({{drvPath}});
+
+        auto accessor = store->getFSAccessor();
+        auto outPathS = store->printStorePath(outPath);
+        if (accessor->stat(outPathS).type != FSAccessor::tRegular)
+            throw Error("'%s' is not a file; an exporter must only create a single file", outPathS);
```

Comment:
Not sure if this restriction is really useful. I can imagine some exporters producing multiple files (e.g. I seem to remember that instance-store AMIs are a directory of files).